src/components/ButtonGroup/ButtonGroup.js

```diff
@@ -0,0 +1,74 @@
+import * as React from 'react';
+import * as PropTypes from 'prop-types';
+import styles from './style.scss';
+
+const baseClass = 'btn-group';
+
+class ButtonGroup extends React.Component {
+  state = {
+    currentIndex: this.props.currentIndex || -1
+  };
+
+  makeClickHandler = index => event => {
+    this.props.onChange(event, index);
+    this.setState({ currentIndex: index });
+  };
+
+  render() {
+    const { size, fullWidth, children, ...restProps } = this.props;
+    const currentIndex =
+      typeof this.props.currentIndex === 'number'
```

Comment:
I think we should force number value in this props

Reply:
The component checks for the type of this prop to determine if it was passed and if it contains valid index value to make a decision if the component should use forced index passed through props or use its internal `state.currentIndex`. Of course, I could check if it is not `undefined` but I don't like to use `undefined` keyword explicit 🤔

Reply:
so, maybe use isNaN? :D

Reply:
but w8, you initialize component with state
```
state = {
    currentIndex: this.props.currentIndex || defaultIndex
  };
```

so do you need this check in render?

Reply:
that was my first idea but then eslint comes on the way 🤷‍♂ 
```
Unexpected use of 'isNaN'.eslint(no-restricted-globals)
```

Reply:
Yep because of updates in props, possibly the `currentIndex` prop could be removed or added after component initialization. Now this component is "always ready to update" 😄. The state initialization using this prop is for properly syncing the starting state value: If I pass for example `currentIndex={2}` and then remove it the component will handle tracking by itself from proper index `2` 🙂

Reply:
I have updated the implementation by using the derived state to handle forcing `currentIndex` value from props. That way this variable in the render is not needed and component always relies on its state (but the state could be changed from props) 🙂